tarofreighter/chain_porter.go

```diff
@@ -252,9 +252,7 @@ func (p *ChainPorter) waitForTransferTxConf(pkg *sendPackage) {
 		outboundPkg.AnchorTxHeightHint, true,
 	)
 	if err != nil {
-		p.cfg.ErrChan <- mkErr("unable to register for tx conf: %v",
-			err)
-		return
+		return mkErr("unable to register for tx conf: %v", err)
```

Comment:
We could extract the `mkErr` helper functions into the `advanceState` method as well. Then we'd have a normal `return fmt.Errorf("...")` here and the logging done in a single place.